x-pack/legacy/plugins/snapshot_restore/public/app/components/policy_form/steps/step_settings.tsx

```diff
@@ -0,0 +1,450 @@
+                            <EuiLink
+                              onClick={() => {
+                                indicesOptions.forEach((option: Option) => {
+                                  option.checked = undefined;
+                                });
+                                updatePolicyConfig({ indices: [] });
+                                setCachedIndices([]);
+                              }}
+                            >
+                              <FormattedMessage
+                                id="xpack.snapshotRestore.policyForm.stepSettings.deselectAllIndicesLink"
+                                defaultMessage="Deselect all"
+                              />
+                            </EuiLink>
+                          ) : (
+                            <EuiLink
+                              onClick={() => {
+                                indicesOptions.forEach((option: Option) => {
+                                  option.checked = 'on';
+                                });
+                                updatePolicyConfig({ indices: [...indices] });
+                                setCachedIndices([]);
+                              }}
+                            >
+                              <FormattedMessage
+                                id="xpack.snapshotRestore.policyForm.stepSettings.selectAllIndicesLink"
+                                defaultMessage="Select all"
+                              />
+                            </EuiLink>
+                          ),
+                      }}
+                    />
+                  ) : null
+                }
+                isInvalid={Boolean(errors.indices)}
+                error={errors.indices}
+              >
+                {selectIndicesMode === 'list' ? (
+                  <EuiSelectable
+                    allowExclusions={false}
+                    options={indicesOptions}
+                    onChange={options => {
+                      const newSelectedIndices: string[] = [];
+                      options.forEach(({ label, checked }) => {
+                        if (checked === 'on') {
+                          newSelectedIndices.push(label);
+                        }
+                      });
+                      setIndicesOptions(options);
+                      updatePolicyConfig({ indices: [...newSelectedIndices] });
+                      setCachedIndices([...newSelectedIndices]);
+                    }}
+                    searchable
+                    height={300}
+                  >
+                    {(list, search) => (
+                      <EuiPanel paddingSize="s" hasShadow={false}>
+                        {search}
+                        {list}
+                      </EuiPanel>
+                    )}
+                  </EuiSelectable>
+                ) : (
+                  <EuiComboBox
+                    options={indices.map(index => ({ label: index }))}
```

Comment:
Here the user has decided to define index patterns to target the indices. It does not make much sense to offer him a way to enter an index name. What do you think?

Also, there is no validation on the pattern being inserted (the same validation we do in CCR when adding follower patterns). Not a blocker, as it would be much more easy to add it with the Form lib once merged.
@cjcenizal this is already the 3rd place we need to insert a pattern + validation, I think we can make it a priority to have a reusable solution 😊 

<img width="424" alt="Screen Shot 2019-08-22 at 16 32 57" src="https://user-images.githubusercontent.com/2854616/63523481-885a5400-c4fa-11e9-8d01-c4f3387d78c2.png">